src/compiler/tsc.ts

```diff
@@ -360,7 +360,7 @@ namespace ts {
             let newFileNames = ts.map(parsedCommandLine.fileNames, compilerHost.getCanonicalFileName);
             let canonicalRootFileNames = ts.map(rootFileNames, compilerHost.getCanonicalFileName);
 
-            if (!arrayStructurallyIsEqualTo(newFileNames, canonicalRootFileNames)) {
+            if (!arrayIsEqualTo(newFileNames.sort(), canonicalRootFileNames.sort())) {

```

Comment:
If so there will be multiple places to add the null check and comparison (because if `newFilenames` and `canonicalFileNames` are both `null` it should also compare the two), which seems redundant. Plus the added optional parameters don't really have side effects to `arrayIsEqualTo`.